cmd/manager/main_test.go

```diff
@@ -192,3 +223,117 @@ func Test_garbageCollectSoftOwnedSecrets(t *testing.T) {
 		})
 	}
 }
+
+func newFakeK8sClientsetWithDiscovery(resources []*metav1.APIResourceList, discoveryError error) kubernetes.Interface {
+	discoveryClient := &fakeDiscovery{
+		resources:                         resources,
+		errServerResourcesForGroupVersion: discoveryError,
+	}
+	client := &fakeClientset{
+		discovery: discoveryClient,
+	}
+	return client
+}
+
+func Test_determineSetDefaultSecurityContext(t *testing.T) {
+	type args struct {
+		setDefaultSecurityContext string
+		clientset                 kubernetes.Interface
+	}
+	tests := []struct {
+		name    string
+		args    args
+		want    bool
+		wantErr bool
+	}{
+		{
+			"auto-detect in openshift cluster does not set security context",
```

Comment:
```suggestion
			"auto-detect in OpenShift cluster does not set security context",
```

and a couple of times below.